test/validation/api/ipsec/ipsec.h

```diff
@@ -42,6 +45,8 @@ struct suite_context_s {
 	odp_queue_t default_queue;
 	odp_queue_t queue;
 	odp_pktio_t pktio;
+	odp_queue_type_t q_type;
+	odp_event_t (*dest_queue_deq)(void);
```

Comment:
I started wondering if the dest_queue_deq function pointer and the corresponding functions are very useful, as opposed to having a single function that checks the dest_queue type, especially since such function basically already exists in the form of recv_event.